### PR TITLE
feat: enforce that PRs have a jira ticket reference

### PR DESCRIPTION
**Pull request by hspitzley-czi**
<details open>
  <summary><a href="https://czi-tech.atlassian.net/browse/CCIE-1548" title="CCIE-1548" target="_blank">CCIE-1548</a></summary>
  <br />
  <table>
    <tr>
      <th>Summary</th>
  <td>Enforce that all PRs to happy have a Jira ticket linked</td>
    </tr>
    <tr>
      <th>Type</th>
      <td>
        Task
        Task
      </td>
    </tr>
    <tr>
      <th>Status</th>
      <td>In Review</td>
    </tr>
  </table>
</details>

---

Adds a Github action to ensure PRs have a valid jira ticket reference in the branch name, PR title, or PR body. If there is a valid jira ticket reference the action will prepend a summary to the PR description, like the one that was automatically added above ⬆️

### CODE REVIEW

File: .github/workflows/jira-pr.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+      - edited
+      - opened
+      - synchronize
+      - reopened
+
+name: Jira PR
+
+jobs:
+  validate_jira_reference:
+    runs-on: ubuntu-latest
```
Comment (jakeyheath) [inline .github/workflows/jira-pr.yaml:13]: can you change to be self-hosted?

**Review by jakeyheath — APPROVED** [review]

File: .github/workflows/jira-pr.yaml (hunk 1 above)
Comment (hspitzley-czi) [reply .github/workflows/jira-pr.yaml:13]:
```suggestion
    runs-on: [ARM64, self-hosted, Linux]
```

Comment (hspitzley-czi) [reply .github/workflows/jira-pr.yaml:13]: @jakeyheath Is that the correct change?

Comment (jakeyheath) [reply .github/workflows/jira-pr.yaml:13]: yup! just make sure this runs okay on ARM, but that is the correct syntax

Comment (hspitzley-czi) [reply .github/workflows/jira-pr.yaml:13]: Sounds good, we'll see it run in a sec once I merge the change

Comment (hspitzley-czi) [reply .github/workflows/jira-pr.yaml]: Seems like it works as expected

Comment (jakeyheath) [reply .github/workflows/jira-pr.yaml]: 🚢

**Pull request merged**